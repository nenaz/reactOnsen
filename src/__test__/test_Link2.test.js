import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Link2 from '../test_component/Link2';
/* global window */

describe('Link2', () => {
    it('should handle the click event', () => {
        window.alert = jest.fn();
        const output = shallow(
            <Link2 title="mockTitle" url="mockUrl" />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
        output.simulate('click');
        expect(window.alert).toHaveBeenCalledWith('Кликнули по ссылке!');
    });

    it('should handle state changes', () => {
        window.alert = jest.fn();
        const output = shallow(
            <Link2 title="mockTitle" url="mockUrl" />
        );
        // expect(shallowToJson(output)).toMatchSnapshot();
        expect(output.state().clicked).toEqual(false);
        output.simulate('click');
        expect(output.state().clicked).toEqual(true);
    });
});