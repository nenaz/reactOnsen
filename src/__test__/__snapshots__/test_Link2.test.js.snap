// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link2 should handle the click event 1`] = `
<a
  href="mockUrl"
  onClick={[Function]}
>
  mockTitle
</a>
`;
